doc/LibcurlTransportAdapter.md

```diff
@@ -0,0 +1,162 @@
+
+#### No intermediate buffer
+
+Older versions of the Azure SDK have worked around the libcurl programing model by using an intermediate buffer to fetch the entire response from the server and let the program continue from step 4 (from the steps mentioned above). By doing this, the maximum download size gets limited to what the system memory can handle. Older SDK versions would support 2GB as the maximum download size.
+ 
+Another approach is to be able to request data from the server starting at some specific offset. Then, downloading big content would be equivalent to sending one request after another, asking for data starting on the last downloaded data. However, this strategy adds unnecessary extra messaging.  Each request is parsed by the server to produce a response that is sent back to the client. The client also parses the response. These server and client parsing would be happening one after each other, so it is not efficient.
+
+### Poor man's substitute
+
+As mentioned before, using libcurl properly means delegating TCP and HTTP entirely to libcurl. In a nutshell, libcurl works as a state machine. Calling [curl_easy_perform(handle)](https://curl.se/libcurl/c/curl_easy_perform.html) is the way to start the machine (step 4 mentioned above). A loop is started and won't be completed until the request (set it up in the handle) is completed. If the request represents a GET operation to download data, and the application is expecting to read the data coming for the server, the handle needs to be configured with a _delegated function_ (a callback) that libcurl will invoke as soon as it has data from the server. Every time the callback is invoked, libcurl will inform how big it is the chunk of data that has arrived from the server, and a pointer to the buffer where data can be read. The next flow would be an approximation summary of what is happening within libcurl while dispatching a GET request:
+ 
+1.	Resolve DNS from request URL.
+2.	Get a network socket from the OS.
+3.	Establish TCP connection to the server.
+4.	Secure connection with TLS when using HTTPS.
+5.	Create an HTTP request from the handle and write it to the network socket.
+6.	Wait for socket to be readable (means server has responded) and get an HTTP status line and headers.
+7. If there's a payload body (data to be downloaded), a loop is started here as:
+*  7.1 Read data from the network socket.
+*  7.2 Invoke the application callback passing the data read from the socket.
+* 7.3 Repeat until the entire response has been received.
+ 
+This flow demonstrates how using libcurl to download data requires developers to use a callback pattern. For the example of reproducing a media file (mentioned above), playing media must happen as part of the callback that is invoked by libcurl, which is very unfortunate. 
```

Comment:
It's more than just "unfortunate" - it's untenable because of the requirements associated with multimedia rendering.